src/lib.rs

```diff
@@ -179,16 +192,35 @@ impl Catalog {
 }
 
 #[derive(Clone, Debug, Eq, PartialEq)]
-struct Message {
-    id: String,
-    context: Option<String>,
-    translated: Vec<String>,
+/// `Message` represents a message that can be translated. It contains
+/// the original string (ID), an optional plural form, an optional context
+/// for disambiguation, and the translated strings for the message.
+pub struct Message {
+    /// The original string to be translated, used as the key for looking up
+    /// translations.
+    pub id: String,
+    /// An optional plural form of the original string, used for languages
+    /// that have more than one form for plurals.
+    pub id_plural: Option<String>,
+    /// An optional context for the translation, used for disambiguation
+    /// when the same original string can have different translations
+    /// depending on its usage.
+    pub context: Option<String>,
+    /// Translated strings for the message. Contains one string for each
+    /// plural form in the target language.
+    pub translated: Vec<String>,
 }
 
 impl Message {
-    fn new<T: Into<String>>(id: T, context: Option<T>, translated: Vec<T>) -> Self {
+    fn new<T: Into<String>>(
+        id: T,
+        id_plural: Option<T>,
```

Comment:
Changing the Message object seems like it might be problematic in terms of API breakage, and technically the plural isn't part of the ID afaik. Could we handle this a different way? Not sure exactly how 🤔

Reply:
@rmccue 

>Changing the Message object seems like it might be problematic in terms of API breakage

Considering the fact that it's an optional value I don't think it would cause API breakage. Are there specific scenarios in which you envisage this happening?

>and technically the plural isn't part of the ID afai

I supposed you are referring to the `id_` prefix, yes it's not part of the ID technically. I've remove the prefix.

FWIW the plural was part of the Message object originally in https://github.com/justinas/gettext/commit/a65e147630198e0cf082d9e17a19816aeeeefb08 but seemed to have been removed in https://github.com/justinas/gettext/commit/ffb2b97d916b748788b26623582181270e458ac6 because no use was found for it as at that time but now we do have a use for it.

Reply:
I have updated the code to move the `plural` to a new associated function `with_plural` within the `Message` object. This ensures that existing API integration to the `Message` object will not break if there are no plural forms and will still be successful if there are plural forms.